src/plugins/plugin.legend.js

```diff
@@ -411,15 +412,30 @@ var Legend = Element.extend({
 			// Horizontal
 			var isHorizontal = me.isHorizontal();
 			if (isHorizontal) {
+				// default to center as this is the original behaviour of just position
+				var alignedX = ((legendWidth - lineWidths[0]) / 2) + labelOpts.padding;
+				if (opts.alignment === 'left') {
+					alignedX = 0;
+				} else if (opts.alignment === 'right') {
+					alignedX = ((legendWidth - lineWidths[0])) + labelOpts.padding;
+				}
 				cursor = {
-					x: me.left + ((legendWidth - lineWidths[0]) / 2) + labelOpts.padding,
+					x: me.left + alignedX,
```

Comment:
As `lineWidths` includes padding on the both sides, we need to add `labelOpts.padding` in any case.

```javascript
				var alignedX = me.left + (legendWidth - lineWidths[0]) / 2;
 				if (opts.alignment === 'left') {
 					alignedX = me.left;
 				} else if (opts.alignment === 'right') {
 					alignedX = me.right - lineWidths[0];
 				}
 				cursor = {
 					x: alignedX + labelOpts.padding,
```